compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -328,6 +328,14 @@ public void visit(BLangPackage pkgNode, AnalyzerData data) {
             }
         }
 
+        // Visiting after defining constants due to possible use of constants within xml namespaceURI.
+        for (int i = 0; i < topLevelNodes.size(); i++) {
+            TopLevelNode node = topLevelNodes.get(i);
+            if (node.getKind() == NodeKind.XMLNS) {
+                analyzeNode((BLangNode) node, data);
+            }
+        }
```

Comment:
Can't we bring this logic into the symbolEnter phase where we define module level constructs?

Consider following case,
```
xmlns CONSTA as ns;
const CONSTA = "example.com"; 
```
If we stick with depth first resolving logic used for other constructs then logic to solve the above will be like, we visit `module-xmlns-decl` to solve that we need to solve `CONSTA` then we solve `CONSTA`(`module-const-decl`) after all we can resolve `module-xmlns-decl`.

We may have to handle it here.
https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeResolver.java#L226

Reply:
+1 for @prakanth97 suggestion. Since in the future we should support the xml template declaration as well according to the spec. In that case we can have namespaces in the xml template.
```ballerina
xmlns "example.com" as ns;
const CONSTA = xml `</ns:a>`
```

Reply:
As discussed offline following the approach described in this comment.